### ISSUE TITLE
groff PDF driver is broken

### ISSUE THREAD

**Issue by Profpatsch** [NixOS/nixpkgs#53056]
With https://github.com/NixOS/nixpkgs/pull/27305, using the pdf driver broke: 

```
> nix-shell -p groff.out groff.perl
[nix-shell:~]$ man --troff-device=pdf man
groff: can't find `DESC' file
groff:fatal error: invalid device `pdf'
/run/current-system/sw/bin/man: command exited with status 3: /nix/store/bdjpyk4h2r9fjj1d8z5c9dfgdr1ydb8k-man-db-2.7.5/libexec/man-db/zsoelim | /nix/store/bdjpyk4h2r9fjj1d8z5c9dfgdr1ydb8k-man-db-2.7.5/libexec/man-db/manconv -f UTF-8:ISO-8859-1 -t UTF-8//IGNORE | preconv -e UTF-8 | tbl | groff -mandoc -Tpdf
```

`strace` tells the reason:

```
[nix-shell:~]$ strace -e trace=file groff -T pdf $(man --where man) 2>&1 | grep DESC
openat(AT_FDCWD, "/nix/store/nsqkggfg6akqpga6im273zdhmisc3pnp-groff-1.22.3/share/groff/site-font/devpdf/DESC", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/nsqkggfg6akqpga6im273zdhmisc3pnp-groff-1.22.3/share/groff/1.22.3/font/devpdf/DESC", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/font/devpdf/DESC", O_RDONLY) = -1 ENOENT (No such file or directory)
groff: can't find `DESC' file
```

The `perl` output is not searched by `groff`.

cc @therealpxc

**Comment by therealpxc** [comment]
Just not moving the TMAC and DESC fixes the command in the description of this issue, but I'm not sure if it breaks some of the other, perl-using commands.

I'm playing with that and other options right now.

**Comment by Profpatsch** [comment]
It’s still broken on master.

**Comment by j0hax** [comment]
@Profpatsch Any solutions yet?

**Comment by Profpatsch** [comment]
Not that I know of

**Comment by pniedzwiedzinski** [comment]
Thanks to @stalebot I reminded myself that this bug sucks! So I decided to hack a little bit and try to build it from source. And it works! Pdf just works!

So then I thought "hmm maybe that's some nix issue with symlinking". So I wrapped everything I did from scratch into this derivation:
```nix
{ stdenv, fetchurl, perl, texinfo }:

stdenv.mkDerivation {
  pname = "groff";
  version = "1.22.4";

  src = fetchurl {
    url = "https://ftp.gnu.org/gnu/groff/groff-1.22.4.tar.gz";
    sha256 = "14q2mldnr1vx0l9lqp9v2f6iww24gj28iyh4j2211hyynx67p3p7";
  };

  buildInputs = [ perl texinfo ];

  configurePhase = ''
    ./configure --prefix=$out
  '';
}
```

:partying_face: and it works. So these are a couple of my ideas of what's happening here:
- either my simple derivation produce that generates pdfs but has some other hidden problems
- or the official derivation is too complicated and there's some hidden error there

I don't really know `groff` and my experience with this program stops with rendering the larbs manual, so I'm posting this here with hope that somebody will test it through and fix the bug :balloon:

**Comment by abathur** [comment]
Closing since it looks like this was fixed in #101866. Happy to reopen if I'm mistaken (but the output I get from `nix-shell --pure -p groff.out groff.perl --command 'man --troff-device=pdf man'` makes me think this is working).
